src/core/js/models/adaptModel.js

```diff
@@ -247,6 +247,64 @@ define([
       Adapt.checkedCompletion();
     }
 
+    /**
+     * Returns a string describing the type group of this model.
+     * Strings should be lowercase and not plurlaized.
+     * i.e. 'page', 'menu', 'contentobject', 'component', 'article', 'block'
+     * Override in inheritance chain.
+     * @returns {string}
+     */
+    getTypeGroup() {}
+
+    /**
+     * Returns true if this model is of the type group described.
+     * Automatically manages pluralization typeGroup and matches lowercase only.
+     * Pluralized typeGroups and uppercase characters in typeGroups are discouraged.
+     * @param {string} type Type group name i.e. course, contentobject, article, block, component
+     * @returns {boolean}
+     */
+    isTypeGroup(typeGroup) {
+      const hasUpperCase = /[A-Z]+/.test(typeGroup);
+      const isPluralized = typeGroup.slice(-1) === 's';
+      const lowerCased = typeGroup.toLowerCase();
+      const singular = isPluralized && lowerCased.slice(0, -1); // remove pluralization if ending in s
+      const singularLowerCased = (singular || lowerCased).toLowerCase();
+      if (isPluralized || hasUpperCase) {
+        const message = (isPluralized && hasUpperCase) ?
+          `'${typeGroup}' appears pluralized and contains uppercase characters, suggest using the singular, lowercase type group '${singularLowerCased}'.` :
+          isPluralized ?
+            `'${typeGroup}' appears pluralized, suggest using the singular type group '${singularLowerCased}'.` :
+            `'${typeGroup}' contains uppercase characters, suggest using lowercase type group '${singularLowerCased}'.`;
+        Adapt.log.deprecated(message);
```

Comment:
Thinking this code is now too much for what is a simple message; would you be happy making it generic and losing the nested logic? Something like:
```js
Adapt.log.deprecated(`'${typeGroup}' appears pluralized or contains uppercase characters, suggest using the singular, lowercase type group '${singularLowerCased}'.`);
```